app/src/main/java/id/rizmaulana/covid19/ui/dailygraph/DailyDataListFragment.kt

```diff
@@ -0,0 +1,62 @@
+package id.rizmaulana.covid19.ui.dailygraph
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.LinearLayoutManager
+import id.rizmaulana.covid19.databinding.FragmentDailyListBinding
+import id.rizmaulana.covid19.ui.adapter.DailyFactoryImpl
+import id.rizmaulana.covid19.ui.adapter.VisitableRecyclerAdapter
+import id.rizmaulana.covid19.ui.base.BaseFragment
+import id.rizmaulana.covid19.util.ext.observe
+import org.koin.android.viewmodel.ext.android.sharedViewModel
+
+class DailyDataListFragment : BaseFragment() {
+
+    private val viewModel by sharedViewModel<DailyGraphViewModel>()
+
+    private var _binding: FragmentDailyListBinding? = null
+    private val binding get() = _binding!!
+    private val dailyAdapter by lazy {
+        VisitableRecyclerAdapter(DailyFactoryImpl())
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        _binding = FragmentDailyListBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        with(binding.recyclerView) {
+            layoutManager = LinearLayoutManager(context)
+            adapter = dailyAdapter
+        }
+        binding.swipeRefresh.setOnRefreshListener {
+            viewModel.loadRemoteDailyData()
+        }
```

Comment:
You can do it like this

```
super.onViewCreated(view, savedInstanceState)

binding = FragmentDailyListBinding.bind(view)
with(binding){
  ....
}
```

Reply:
> You can do it like this
> 
> ```
> super.onViewCreated(view, savedInstanceState)
> 
> binding = FragmentDailyListBinding.bind(view)
> with(binding){
>   ....
> }
> ```

hmm yeah, i think both way are equally good, im just following official docs on fragment, so i think thats okay
https://developer.android.com/topic/libraries/view-binding